internal/server/singleprocess/service_job.go

```diff
@@ -279,6 +283,10 @@ func (s *service) onDemandRunnerStartJob(
 ) (*pb.Job, string, error) {
 	log := hclog.FromContext(ctx)
 
+	if od == nil {
+		return nil, "", status.Error(codes.FailedPrecondition, "Missing required parameter OnDemandRunnerConfig")
+	}
+
```

Comment:
Nit: I don't see any other way into this method outside of the above `wrapJobWithRunner`, which already performs a `nil` check on the `OnDemandRunnerConfig`, so this seems redundant.

Reply:
Came to say the same thing :)